packages/generator/src/sdk-metadata/generation-and-usage.ts

```diff
@@ -10,41 +17,61 @@ export async function getGenerationAndUsage(
     genericUsage: await getGenericUsage(),
     apiSpecificUsage: await getApiSpecificUsage(service),
     links: getLinks(),
-    generationSteps: getGenerationDocumentation()
+    generationSteps: getGenerationSteps(),
+    generatorVersion: getGeneratorVersion(),
+    generatorRepositoryLink:
+      'https://www.npmjs.com/package/@sap-cloud-sdk/generator'
   };
 }
 
-export async function getGenericUsage(): Promise<string> {
-  return genericGetAllCodeSample(
-    'BusinessPartner',
-    '@sap/cloud-sdk-vdm-business-partner-service'
-  );
+export async function getGenericUsage(): Promise<
+  InstructionWithText<typeof genericUsageText>
+> {
+  return {
+    instructions: genericGetAllCodeSample(
+      'BusinessPartner',
+      '@sap/cloud-sdk-vdm-business-partner-service'
+    ),
+    text: genericUsageText
+  };
 }
 
+export const apiSpecificUsageText = 'Find below a usage example for this API.';
 export async function getApiSpecificUsage(
   service: VdmServiceMetadata
-): Promise<string> {
+): Promise<InstructionWithText<typeof apiSpecificUsageText>> {
```

Comment:
I am not sure whether it is a good idea to have the usage string as literal type here... Why not just string?

Reply:
hm I wanted to be super rigorous, but I think you are right. I make it just string.